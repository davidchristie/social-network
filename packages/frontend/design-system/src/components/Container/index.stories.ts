import Container from ".";
import addComponentStory from "../../utilities/storybook/addComponentStory";

addComponentStory(Container);
