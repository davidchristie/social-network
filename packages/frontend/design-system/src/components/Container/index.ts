import elementWithClass from "../../utilities/elementWithClass";
import "./index.css";

interface Props extends React.HTMLAttributes<HTMLElement> { }

const Container = elementWithClass<Props>("Container", "div");

export default Container;
