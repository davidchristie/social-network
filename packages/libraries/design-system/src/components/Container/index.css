.Container {
  margin-left: auto;
  margin-right: auto;
  width: 100%;
}

@media (min-width: 575px) {
  .Container {
    max-width: 550px;
  }
}

@media (min-width: 775px) {
  .Container {
    max-width: 720px;
  }
}

@media (min-width: 1000px) {
  .Container {
    max-width: 950px;
  }
}

@media (min-width: 1200px) {
  .Container {
      max-width: 1150px;
  }
}
