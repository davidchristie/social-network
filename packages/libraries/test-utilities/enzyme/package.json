{
  "main": "../dist/enzyme/index.js",
  "name": "test-utilities/enzyme",
  "private": true,
  "version": "1.0.0"
}
