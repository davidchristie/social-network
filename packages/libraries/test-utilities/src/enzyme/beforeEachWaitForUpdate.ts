import { Wrapper } from "./types";

export default function beforeEachWaitForUpdate (getWrapper: () => Wrapper) {
  beforeEach(async () => {
    await new Promise(resolve => window.setTimeout(resolve, 0));
    getWrapper().update();
  });
}
