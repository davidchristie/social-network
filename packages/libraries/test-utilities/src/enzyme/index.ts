export { default as beforeEachWaitForUpdate } from "./beforeEachWaitForUpdate";
export { default as itContainsComponent } from "./itContainsComponent";
