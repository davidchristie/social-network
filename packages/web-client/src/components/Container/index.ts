import divWithClass from "../../factories/divWithClass";
import "./index.css";

export default divWithClass("Container");
