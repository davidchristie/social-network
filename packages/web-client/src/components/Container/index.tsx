import React from "react";

import "./index.css";

type Props = React.HTMLAttributes<HTMLDivElement>;

export default class Container extends React.Component<Props> {
  public render () {
    const { children, className, ...attributes } = this.props;
    return (
      <div
        className={`Container${className ? " " + className : ""}`}
        {...attributes}
      >
        {children}
      </div>
    );
  }
}
