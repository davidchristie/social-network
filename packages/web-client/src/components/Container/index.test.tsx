import { shallow, ShallowWrapper } from "enzyme";
import React from "react";

import Container from "./index";

describe("Container component", () => {
  let wrapper: ShallowWrapper;

  describe("with no props", () => {
    beforeEach(() => {
      wrapper = shallow(<Container />);
    });

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("with custom class", () => {
    const CUSTOM_CLASS = "custom_class";

    beforeEach(() => {
      wrapper = shallow(
        <Container className={CUSTOM_CLASS} />,
      );
    });

    it("has custom class", () => {
      expect(wrapper.hasClass(CUSTOM_CLASS)).toBe(true);
    });

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
