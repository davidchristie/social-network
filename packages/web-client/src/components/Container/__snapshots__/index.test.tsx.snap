// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component with custom class matches snapshot 1`] = `
<div
  className="Container custom_class"
/>
`;

exports[`Container component with no props matches snapshot 1`] = `
<div
  className="Container"
/>
`;
